src/export-grants.ts

```diff
@@ -21,40 +22,85 @@ function getTxAmount(decodedLogEvent: Decoded, decimals: number) {
+
+  const raw_vesting_releasable = await vestingContract.methods.getReleasable().call()
+  const vesting_releasable = parseNumber(raw_vesting_releasable, decimals)
+
+  const contractStart: number = await vestingContract.methods.getStart().call()
+  const contractDuration: number = await vestingContract.methods.getPeriod().call()
+  const contractEndsTimestamp = Number(contractStart) + Number(contractDuration)
+  const vesting_start_at = toISOString(contractStart)
+  const vesting_finish_at = toISOString(contractEndsTimestamp)
+
+  const tokenContract = new web3.eth.Contract(token.abi, contract_token_address)
+  const raw_token_contract_balance = await tokenContract.methods.balanceOf(vestingAddress).call()
+  const vesting_token_contract_balance = parseNumber(raw_token_contract_balance, decimals)
+  const vesting_total_amount = vesting_token_contract_balance + vesting_released
+
+  return {
+    token: token.symbol,
+    vesting_released,
+    vesting_releasable,
+    vesting_start_at,
+    vesting_finish_at,
+    vesting_token_contract_balance,
+    vesting_total_amount
+  }
+}
+
 async function getVestingContractData(proposalId: string, vestingAddress: string): Promise<VestingInfo> {
   try {
-    const vestingContract = new web3.eth.Contract(VESTING_ABI as AbiItem[], vestingAddress)
-    const contract_token_address: string = (await vestingContract.methods.token().call()).toLowerCase()
-    const token = Tokens.getEthereumToken(contract_token_address)
-    const decimals = token.decimals
-
-    const raw_vesting_released = await vestingContract.methods.released().call()
-    const vesting_released = parseNumber(raw_vesting_released, decimals)
-
-    const raw_vesting_releasable = await vestingContract.methods.releasableAmount().call()
-    const vesting_releasable = parseNumber(raw_vesting_releasable, decimals)
-
-    const contractStart: number = await vestingContract.methods.start().call()
-    const contractDuration: number = await vestingContract.methods.duration().call()
-    const contractEndsTimestamp = Number(contractStart) + Number(contractDuration)
-    const vesting_start_at = toISOString(contractStart)
-    const vesting_finish_at = toISOString(contractEndsTimestamp)
-
-    const tokenContract = new web3.eth.Contract(token.abi, contract_token_address)
-    const raw_token_contract_balance = await tokenContract.methods.balanceOf(vestingAddress).call()
-    const vesting_token_contract_balance = parseNumber(raw_token_contract_balance, decimals)
-    const vesting_total_amount = vesting_token_contract_balance + vesting_released
-
-    return {
-      token: token.symbol,
-      vesting_released,
-      vesting_releasable,
-      vesting_start_at,
-      vesting_finish_at,
-      vesting_token_contract_balance,
-      vesting_total_amount
-    }
+    return await _getVestingContractDataV1(vestingAddress)
   } catch (e) {
+    // do nothing
+  }
+
+  try {
+    return await _getVestingContractDataV2(vestingAddress)
```

Comment:
Isn't there a way to know which endpoint should we use before calling the API?

If not, I think it's better if this is inside the previous catch (and if we make sure the error is the expected one)

```typescript
try {
  v1
} catch(e){
  check error
  try {
    v2
  catch(e) {
    report to rollbar!
   }
}
```

But I'd much rather have something like

```typescript
try {
  question ? v1 : v2
} catch(e){
  report to rollbar!
}
```

Reply:
As far as I know, the only way to detect which version of vesting is being used is if the contracts had a method to return the version, which they don't have.
I looked if there were alternatives but I did not find any other than that.

I will include Rollbar!